### PR TITLE
Update Lerp op

### PR DESCRIPTION
**Pull request by umadevimcw**
### Ticket
#13625 

### Problem description
Lerp throws error in mixed dtype

### What's changed
- Torch lerp op needs end and weight tensors to be of same dtype as input dtype. An error was thrown from torch call. Hence updated the golden function accordingly
- Added TT_FATAL for lerp op
- Updated in the doc

image

image

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/11856547160)
- [x] [Sweep tests](https://github.com/tenstorrent/tt-metal/actions/runs/11855707975)

### CODE REVIEW

**Review by mcw-anasuya — APPROVED** [review]

**Review by VirdhatchaniKN — APPROVED** [review]

File: tests/sweep_framework/sweeps/eltwise/ternary/lerp/lerp.py [hunk 1]
```diff
@@ -42,6 +42,18 @@
...
 
 
+# Invalidate vector is called during the generation phase where each vector will be passed in.
+# If invalidated, the vector will still be stored but will be skipped.
+# Returns False, None if the vector is valid, and True, str with a reason for invalidation if it is invalid.
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    if (
+        test_vector["input_b_dtype"] != test_vector["input_a_dtype"]
+        or test_vector["input_c_dtype"] != test_vector["input_a_dtype"]
+    ):
```
Comment (Aswinmcw) [inline tests/sweep_framework/sweeps/eltwise/ternary/lerp/lerp.py:52]:
```suggestion
    if test_vector["input_a_dtype"] not in (test_vector["input_b_dtype"], test_vector["input_c_dtype"]):
```

**Review by Aswinmcw — APPROVED** [review]
LGTM

File: tests/sweep_framework/sweeps/eltwise/ternary/lerp/lerp.py (hunk 1 above)
Comment (umadevimcw) [reply tests/sweep_framework/sweeps/eltwise/ternary/lerp/lerp.py:52]: Actually, both the input tensors b and c should be of the same type as input tensor a, hence used relational operator

**Review by KalaivaniMCW — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by jdesousa-TT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]
@umadevimcw ,  
Can you please update the variable names to align?

Parameters
input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the tensor with the starting points

end ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the tensor with the ending points

weight ([float](https://docs.python.org/3/library/functions.html#float) or tensor) – the weight for the interpolation formula

**Comment by umadevimcw** [comment]
> @umadevimcw , Can you please update the variable names to align?
> 
> Parameters input ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the tensor with the starting points
> 
> end ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the tensor with the ending points
> 
> weight ([float](https://docs.python.org/3/library/functions.html#float) or tensor) – the weight for the interpolation formula

image

Updated @eyonland

**Pull request merged**